moped-editor/src/views/projects/projectView/SubprojectFundingModal.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Box,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  IconButton,
+  Typography,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import AddCircle from "@material-ui/icons/AddCircle";
+import { useSocrataJson } from "src/utils/socrataHelpers";
+import MaterialTable from "@material-table/core";
+
+import typography from "../../../theme/typography";
+
+const useStyles = makeStyles((theme) => ({
+  dialogTitle: {
+    color: theme.palette.primary.main,
+    fontFamily: theme.typography.fontFamily,
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+}));
+
+const SubprojectFundingModal = ({
+  isDialogOpen,
+  handleDialogClose,
+  eCaprisID,
+  fdusArray,
+  addProjectFunding,
+  userId,
+  projectId,
+  setSnackbarState,
+}) => {
+  const classes = useStyles();
+  const typographyStyle = {
+    fontFamily: typography.fontFamily,
+    fontSize: "14px",
+  };
+
+  const { data } = useSocrataJson(
+    `https://data.austintexas.gov/resource/jega-nqf6.json?dept_unit_status=Active&sp_number_txt=${eCaprisID}&$limit=9999`
+  );
+
+  const [selectedFdus, setSelectedFdus] = useState([]);
+
+  // Filter the list of fdus to remove one(s) already on funding sources table
+  const filteredData = data.filter((fdu) => !fdusArray.includes(fdu.fdu));
+
+  const columns = [
+    {
+      field: "fdu",
+      title: "FDU",
+      cellStyle: { padding: "12px" },
+    },
+    {
+      field: "unit_long_name",
+      title: "Unit name",
+      cellStyle: { padding: "12px" },
+    },
+  ];
+
+  const handleAddFunding = () => {
+    const newFunds = [];
+    // format record to match generic records added
+    selectedFdus.forEach((fdu) => {
+      const fduRecord = {};
+      fduRecord.dept_unit = {
+        dept: fdu.dept,
+        dept_id: fdu.dept_id,
+        dept_unit_id: fdu.dept_unit_id,
+        dept_unit_status: fdu.dept_unit_status,
+        unit: fdu.unit,
+        unit_long_name: fdu.unit_long_name,
+        unit_short_name: fdu.unit_short_name,
+      };
+      fduRecord.fund = {
+        fund_id: fdu.fund,
+        fund_name: fdu.fundname,
+      };
```

Comment:
i was just writing similar code in the Data & Performance hub. i messed around with fancy destructing, but it's all pretty gross. if we already have lodash in the project, [`pick`](https://lodash.com/docs/4.17.15#pick) would make this cleaner, but i do think this verbose way makes it's very clear whats going on and is good.